aea/helpers/base.py

```diff
@@ -72,27 +73,63 @@ def locate(path):
     return object
 
 
-def load_module(name: str, filepath: os.PathLike):
+def load_module(dotted_path: str, filepath: os.PathLike):
     """
     Load a module.
 
-    :param name: the name of the package/module.
+    :param dotted_path: the dotted path of the package/module.
     :param filepath: the file to the package/module.
     :return: None
     :raises ValueError: if the filepath provided is not a module.
     :raises Exception: if the execution of the module raises exception.
     """
-    spec = importlib.util.spec_from_file_location(name, filepath)
+    spec = importlib.util.spec_from_file_location(dotted_path, filepath)
     module = importlib.util.module_from_spec(spec)
     spec.loader.exec_module(module)  # type: ignore
     return module
 
 
-def add_agent_component_module_to_sys_modules(item_type: str, item_name: str, module_obj):
-    """Add an agent component module to sys.modules."""
+def add_module_to_sys_modules(dotted_path: str, module_obj) -> None:
+    """
+    Add module to sys.modules.
+
+    :param dotted_path: the dotted path to be used in the imports.
+    :param module_obj: the module object. It is assumed it has been already executed.
+    :return: None
+    :raises ValueError: if the dotted path is already used. # TODO this cannot be done atm.
+    """
+    # if dotted_path in sys.modules:
```

Comment:
Why is this commented out?

Reply:
because I wanted feedback :-) 

Those two line should be there for "safety" purposes. That is, to avoid removing previously loaded modules, that could break the code working properly.
In practice, for a (single) AEA, this function shouldn't be a big problem since the namespace of the modules is very narrow (e.g. unlikely that there exist `packages.package_type.package_name`). Still, name conflicts may occur.

A practical consequence of those two lines is that tests fail because each test is not independent in that sense - later test try to load a package which has been already loaded by previous tests, causing the commented `ValueError`  to be raised.